### PR TITLE
Initial Device Tree for Kenya and Nigeria Platforms

### PR DESCRIPTION
**Pull request by modolaty**
Added Kenya and Nigeria Device Trees

### CODE REVIEW

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts [hunk 1]
```diff
@@ -0,0 +1,296 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2024 AMD Inc.
+// Author: Mahesh Kurapati <mahesh.kurapati@amd.com>
```
Comment (supven01) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts:3]: Mo - You can add your name here as well.

**Review by supven01 — APPROVED** [review]

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts [hunk 2]
```diff
@@ -0,0 +1,296 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2024 AMD Inc.
+// Author: Mahesh Kurapati <mahesh.kurapati@amd.com>
```
Comment (supven01) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts:3]: You can add your name as well.

**Review by supven01 — APPROVED** [review]

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+		video_engine_memory: jpegbuffer {
+			size = <0x02000000>;	/* 32M */
+			alignment = <0x01000000>;
+			compatible = "shared-dma-pool";
+			reusable;
+		};
+
+		pcc_memory: pccbuffer {
+                        reg = <0xE0000000 0x00001000>; /* 4K */
```
Comment (rajaganeshr) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts]: AST2700 is 64-bit addressing.
so, all memory register are twice the length. (two 32-bit register values)
i think this should be 
"reg = <0x4 0xE0000000 0x00001000>; /* 4K */" 

something Like line 27:
	memory@400000000 {
		device_type = "memory";
		reg = <0x4 0x00000000 0x0 0x80000000>;
	};

**Review by rajaganeshr — CHANGES_REQUESTED** [review]

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts (hunk 3 above)
Comment (ojayanth) [reply arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts]: this is not required in the latest driver , please remove.

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+			reg = <0x4 0x30000000 0x0 0x80000>;
+			no-map;
+		};
+
+		optee_core: optee_core@430080000 {
+			reg = <0x4 0x30080000 0x0 0x1000000>;
+			no-map;
+		};
+
+		video_engine_memory: jpegbuffer {
```
Comment (ojayanth) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts]: Changes required based on sdk , please refer the PR related to this for details.

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+			reg = <0x4 0x30000000 0x0 0x80000>;
+			no-map;
+		};
+
+		optee_core: optee_core@430080000 {
+			reg = <0x4 0x30080000 0x0 0x1000000>;
+			no-map;
+		};
+
+		video_engine_memory: jpegbuffer {
```
Comment (ojayanth) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts]: Changes required based on sdk , please refer the PR related to this for details.

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+			size = <0x02000000>;	/* 32M */
+			alignment = <0x01000000>;
+			compatible = "shared-dma-pool";
+			reusable;
+		};
+
+		pcc_memory: pccbuffer {
```
Comment (ojayanth) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts]: Not required please remove

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+	perif-dma-mode;
+	oob-dma-mode;
+	flash-dma-mode;
+};
+
+&lpc0_pcc {
+        port-addr = <0x80>;
+        dma-mode;
+        A2600-15;
+        memory-region = <&pcc_memory>;
```
Comment (ojayanth) [inline arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts]: Not required  for latest driver please remove.

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts (hunk 3 above)
Comment (modolaty) [reply arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-kenya.dts]: Done.
Removed pcc_memory

File: arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts (hunk 5 above)
Comment (modolaty) [reply arch/arm64/boot/dts/aspeed/aspeed-bmc-amd-nigeria.dts]: done

**Review by supven01 — APPROVED** [review]

**Review by rajaganeshr — APPROVED** [review]

**Pull request merged**